src/net/ifaddrs/tests.rs

```diff
@@ -0,0 +1,76 @@
+#![cfg(test)]
+
+use std::ptr::null_mut;
+use std::mem::transmute;
+
+use std::ffi::CString;
+
+use libc;
+use libc::{ifaddrs, sockaddr, sockaddr_in};
+use libc::in_addr;
+
+use super::iff_flags::*;
+use super::InterfaceAddrs;
+use super::InterfaceMap;
+
+// Utility function to turn some bytes into a u32 in the right order because
+// endianness is a pain
+fn convert_v4_addr(b0: u8, b1: u8, b2: u8, b3: u8) -> in_addr {
+    unsafe { in_addr { s_addr: transmute([b0, b1, b2, b3]) } }
+}
+
+// Utility function to turn some bytes into a sockaddr with the v4 addr type
+fn sockaddr_for_addr(b0: u8, b1: u8, b2: u8, b3: u8) -> sockaddr {
+    let sin = sockaddr_in {
+        sin_family: libc::AF_INET as u16,
+        sin_port: 0,
+        sin_addr: convert_v4_addr(b0, b1, b2, b3),
+        sin_zero: [0; 8],
+    };
+
+    unsafe { transmute(sin) }
+}
+
+#[test]
+fn test_ifaddrs() {
```

Comment:
An overall summary of this test would be useful as a comment above it (put it above the `#[test]` part)